src/Mono.Android/Xamarin.Android.Net/AndroidClientHandler.Legacy.cs

```diff
@@ -361,7 +361,13 @@ Task ConnectAsync (HttpURLConnection httpConnection, CancellationToken ct)
 		protected virtual async Task WriteRequestContentToOutput (HttpRequestMessage request, HttpURLConnection httpConnection, CancellationToken cancellationToken)
 		{
 			using (var stream = await request.Content.ReadAsStreamAsync ().ConfigureAwait (false)) {
-				await stream.CopyToAsync(httpConnection.OutputStream!, 4096, cancellationToken).ConfigureAwait(false);
+				cancellationToken.ThrowIfCancellationRequested ();
+				try {
+					await stream.CopyToAsync(httpConnection.OutputStream!, 4096, cancellationToken).ConfigureAwait(false);
+				} catch (ObjectDisposedException ex) {
+					Logger.Log (LogLevel.Error, LOG_APP, $"Stream disposed while copying the content stream to output: {ex}");
+					cancellationToken.ThrowIfCancellationRequested ();
```

Comment:
I think it's fine, since the method returns a `Task` and, thus, all the exceptions are observable - but it might not be the case if the thread is being cancelled.